easyaccess/eautils/dtypes.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+"""
+Module for dealing with data type conversion between
+Oracle, python, numpy, FITS, pandas, ...
+
+Some useful documentation:
+Oracle: https://docs.oracle.com/cd/B28359_01/server.111/b28318/datatype.htm
+cx_Oracle: https://cx-oracle.readthedocs.org/en/latest/
+numpy: http://docs.scipy.org/doc/numpy/reference/arrays.dtypes.html
+FITS: https://heasarc.gsfc.nasa.gov/docs/software/fitsio/c/c_user/node20.html
+
+Also, can double check agains Erin Sheldon's DESDB module:
+https://github.com/esheldon/desdb
+
+Numpy datetimes:
+http://docs.scipy.org/doc/numpy-1.10.0/reference/arrays.datetime.html
+"""
+
+import cx_Oracle
+import numpy as np
+
+# Oracle data types
+or_n  = cx_Oracle.NUMBER
+or_s  = cx_Oracle.STRING
+or_f  = cx_Oracle.NATIVE_FLOAT
+or_dt = cx_Oracle.DATETIME
+or_ts = cx_Oracle.TIMESTAMP
+# This is actually OBJECTVAR (hence 'or_ov')
+or_ov  = cx_Oracle.OBJECT
+
+def oracle2numpy(desc):
+    """Takes an Oracle data type and converts to a numpy dtype string.
+
+    TODO: Vectorize?
+    
+    Parameters:
+    ----------
+    info: Oracle column descriptor
+
+    Returns:
+    dtype: Numpy dtype string
+    """
+    name = desc[0]
+    otype = desc[1]
+    size = desc[3]
+    digits = desc[4]
+    scale = desc[5]
+    
+    if otype == or_n:
+        if scale == 0:
+            # Nothing after the decimal; integers
+            if digits == 0:
+                return "i8"
+            elif digits <= 4:
+                return "i2"
+            elif digits <= 9:
+                return "i4"
+            elif digits <= 18:
```

Comment:
I think there is a gap here. We cast int64 to NUMBER(22,0) types during upload. When we try to convert them back they are failing this test (since digits = 22) and being converted to floats. Since we are being sloppy already, it'd be good to at least be consistently sloppy. I'd suggest either:
* Use `digits <= 22` here
* Convert i8 numpy type to NUMBER(19,0) and change this to `digits <= 19`
In either case we'd probably need to think about unsigned ints specially (currently grouped with ints)